src/infer_response.h

```diff
@@ -118,10 +119,18 @@ class InferenceResponseFactory {
   std::function<void(std::unique_ptr<InferenceResponse>&&, const uint32_t)>
       response_delegator_;
 
+
 #ifdef TRITON_ENABLE_TRACING
   // Inference trace associated with this response.
   std::shared_ptr<InferenceTraceProxy> trace_;
 #endif  // TRITON_ENABLE_TRACING
+
+  // The internal unique ID of the request
+  uint64_t request_id_;
+
+  // Response index
+  uint64_t total_response_idx_;
+  std::mutex mu_;
```

Comment:
Should comment here what does this mutex protects.

Reply:
Updated, to use atomic variable.